jtlibrary/python/jtlibrary/src/jtlib/segmentation.py

```diff
@@ -316,19 +311,38 @@ def separate_clumped_objects(clumps_image, min_cut_area, min_area, max_area,
             regions = mh.cwatershed(np.invert(dist), peaks)
 
             # Use the line separating watershed regions to make the cut
-            se = np.ones((3,3), np.bool)
             line = mh.labeled.borders(regions, NEIGHBORHOOD8)
             line[~obj_image] = 0
-            line = mh.morph.dilate(line, NEIGHBORHOOD8)
 
             # Ensure that cut is reasonable given user-defined criteria
             test_cut_image = obj_image.copy()
             test_cut_image[line] = False
-            subobjects, n_subobjects = mh.label(test_cut_image)
+            subobjects, n_subobjects = mh.label(test_cut_image, NEIGHBORHOOD8)
             sizes = mh.labeled.labeled_size(subobjects)
             smaller_object_area = np.min(sizes)
-            smaller_id = np.where(sizes == smaller_object_area)[0][0]
-            smaller_object = subobjects == smaller_id
+
+            # Deal with an edge-case: If trimming is active & there are more
+            # than 2 objects created by the cut, check if they are very small.
+            # If so, remove them.
+            if allow_trimming:
+                if n_subobjects > 2:
+                    if smaller_object_area < trimming_threshold:
+                        tiny_objects = list(np.where(sizes < trimming_threshold)[0])
+                        # Remove objects by adding them to the cutting line
+                        for trim_obj in tiny_objects:
+                            line[subobjects == trim_obj] = True
+                            logger.debug('Trimming an object of size: {}'.format(sizes[trim_obj]))
+
+                        # Redo calculation if split should be applied
+                        test_cut_image = obj_image.copy()
+                        test_cut_image[line] = False
+                        subobjects, n_subobjects = mh.label(test_cut_image,
```

Comment:
Could this produce again sub-objects that are smaller than the threshold in size?  If yes, then you might want to apply 330--342 again (and again); consider changing the `if` at line 327 with a `while`.